### DISCUSSION TITLE
Brendel Bethge Attack Assertion Error

### DISCUSSION THREAD

**Discussion by ApurvaDani** [Trusted-AI/adversarial-robustness-toolbox#1464] [Q&A]
Trying to generate adversarial samples for MNIST Dataset. Facing assertion error while generating the examples. Is this a bug in the library function or why is this happening?

```

AssertionError                            Traceback (most recent call last)
<ipython-input-27-7cb7d5daff2e> in <module>()
      1 attack = BrendelBethgeAttack(estimator=classifier, batch_size=256)
----> 2 x_test_adv = attack.generate(x=X_test_np)
      3 x_test_adv = torch.from_numpy(x_test_adv).to(DEVICE)

/usr/local/lib/python3.7/dist-packages/art/attacks/evasion/brendel_bethge.py in generate(self, x, y, **kwargs)
   2243             assert (np.argmax(self.estimator.predict(x=best_advs), axis=1) == np.argmax(y, axis=1)).all()
   2244         else:
-> 2245             assert (np.argmax(self.estimator.predict(x=best_advs), axis=1) != np.argmax(y, axis=1)).all()
   2246 
   2247         # perform binary search to find adversarial boundary

AssertionError: 
```

**Comment by beat-buesser** [comment]
Hi @ApurvaDani Thank you very much for using ART! I think this assertion error means that at least one of the starting points found by the attack is not adversarial (meaning prediction different than the label). I see you are running `x_test_adv = attack.generate(x=X_test_np)` which means  you are attacking the models predictions. Does this also happen if you attack the true labels with something like `x_test_adv = attack.generate(x=X_test_np, y=Y_test_np_true)`?
